floss/language/rust/decode_utf8.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+import sys
+import logging
+import pathlib
+import argparse
+from typing import Any, List, Tuple, Iterable, Optional
+
+import pefile
+
+MIN_STR_LEN = 4
+
+logger = logging.getLogger(__name__)
+
+
+def get_rdata_section(pe: pefile.PE) -> pefile.SectionStructure:
+    for section in pe.sections:
+        if section.Name.startswith(b".rdata\x00"):
+            return section
+
+    raise ValueError("no .rdata section found")
+
+
+def extract_utf8_strings_from_buffer(buf, min_length=MIN_STR_LEN) -> List[List[Any]]:
+    """
+    Extracts UTF-8 strings from a buffer.
+    """
+
+    # Reference: https://en.wikipedia.org/wiki/UTF-8
+
+    character_and_index = []
+    strings = []
+
+    for i in range(0, len(buf)):
+        # for 1 byte
+        if buf[i] & 0x80 == 0x00:
+            character = buf[i].to_bytes(1, "big").decode("utf-8", "ignore")
+            character_and_index.append([character, i, 1])
```

Comment:
Recommend using a namedtuple or dataclass to capture the character/index/length information in a structured way. Using a list with implicit indices is difficult to follow.